src/graphql/data/nft/Details.ts

```diff
@@ -0,0 +1,160 @@
+import { parseEther } from '@ethersproject/units'
+import graphql from 'babel-plugin-relay/macro'
+import { CollectionInfoForAsset, GenieAsset, Rarity, SellOrder, TokenType } from 'nft/types'
+import { loadQuery, usePreloadedQuery } from 'react-relay'
+
+import RelayEnvironment from '../RelayEnvironment'
+import { DetailsQuery } from './__generated__/DetailsQuery.graphql'
+
+const detailsQuery = graphql`
+  query DetailsQuery($address: String!, $tokenId: String!) {
+    nftAssets(address: $address, filter: { listed: false, tokenIds: [$tokenId] }) {
+      edges {
+        node {
+          id
+          name
+          ownerAddress
+          image {
+            url
+          }
+          smallImage {
+            url
+          }
+          originalImage {
+            url
+          }
+          tokenId
+          description
+          animationUrl
+          suspiciousFlag
+          creator {
+            address
+            profileImage {
+              url
+            }
+            isVerified
+          }
+          collection {
+            name
+            isVerified
+            numAssets
+            image {
+              url
+            }
+            nftContracts {
+              address
+              standard
+            }
+            description
+          }
+          listings(first: 1) {
+            edges {
+              node {
+                address
+                createdAt
+                endAt
+                id
+                maker
+                marketplace
+                marketplaceUrl
+                orderHash
+                price {
+                  currency
+                  value
+                }
+                quantity
+                startAt
+                status
+                taker
+                tokenId
+                type
+                protocolParameters
+              }
+              cursor
+            }
+          }
+          rarities {
+            provider
+            rank
+            score
+          }
+          metadataUrl
+          traits {
+            name
+            value
+          }
+        }
+      }
+    }
+  }
+`
+
+export function useDetailsQuery(address: string, tokenId: string): [GenieAsset, CollectionInfoForAsset] | undefined {
+  const detailsQueryReference = loadQuery<DetailsQuery>(RelayEnvironment, detailsQuery, {
```

Comment:
You should not be using `loadQuery` directly, as you'll then need to deal with disposing of it as well.

In this case, it looks like you may want `useLazyLoadQuery` (instead of `loadQuery`/`usePreloadedQuery`) because `usePreloadedQuery` would put the component in suspense anyways.

In most cases, you probably want `useQueryLoader` at the top-level parent (probably the `pages/<PAGE_NAME>/index.tsx` file), and then `usePreloadedQuery` within the component using the data, so that only that component is suspended should the query not yet be loaded.